lib/Word_Lib/Bitwise.thy

```diff
@@ -165,22 +166,22 @@ lemma rbl_sshiftr:
       drop_nonempty_def take_Cons')
    apply (case_tac "LENGTH('a)", simp_all)
   apply (rule word_eqI)
-  apply (simp add: nth_sshiftr word_size test_bit_of_bl
+  apply (simp add: bit_simps word_size test_bit_of_bl
       msb_nth)
   done
 
 lemma nth_word_of_int:
-  "(word_of_int x :: 'a::len word) !! n = (n < LENGTH('a) \<and> bin_nth x n)"
+  "bit (word_of_int x :: 'a::len word) n = (n < LENGTH('a) \<and> bit x n)"
```

Comment:
we're losing `!!` syntax entirely?

Reply:
It's recreated later again as optional syntax. I don't really like it either, but at least optional does make sense. In our proofs nothing changes in the syntax.